server/rest/src/main/java/org/infinispan/rest/resources/LoggingResource.java

```diff
@@ -0,0 +1,177 @@
+package org.infinispan.rest.resources;
+
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.infinispan.commons.dataconversion.MediaType.APPLICATION_JSON_TYPE;
+import static org.infinispan.rest.framework.Method.DELETE;
+import static org.infinispan.rest.framework.Method.GET;
+import static org.infinispan.rest.framework.Method.PUT;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.concurrent.CompletionStage;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.core.Appender;
+import org.apache.logging.log4j.core.LoggerContext;
+import org.apache.logging.log4j.core.config.LoggerConfig;
+import org.infinispan.rest.InvocationHelper;
+import org.infinispan.rest.NettyRestResponse;
+import org.infinispan.rest.framework.ResourceHandler;
+import org.infinispan.rest.framework.RestRequest;
+import org.infinispan.rest.framework.RestResponse;
+import org.infinispan.rest.framework.impl.Invocations;
+import org.infinispan.tasks.TaskContext;
+import org.infinispan.tasks.TaskManager;
+
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.std.StdSerializer;
+
+import io.netty.handler.codec.http.HttpResponseStatus;
+
+/**
+ * Logging configuration resource. This resource is log4j 2.x-specific
+ *
+ * @author tristan@infinispan.org
+ * @since 11.0
+ */
+public final class LoggingResource implements ResourceHandler {
+   private final InvocationHelper invocationHelper;
+
+   public LoggingResource(InvocationHelper invocationHelper) {
+      this.invocationHelper = invocationHelper;
+      // Register our custom serializers
+      SimpleModule module = new SimpleModule();
+      module.addSerializer(LoggerConfig.class, new Log4j2LoggerConfigSerializer());
+      module.addSerializer(Appender.class, new Log4j2AppenderSerializer());
+      this.invocationHelper.getMapper().registerModule(module);
+   }
+
+   @Override
+   public Invocations getInvocations() {
+      return new Invocations.Builder()
+            .invocation().methods(GET).path("/v2/logging/loggers").handleWith(this::listLoggers)
```

Comment:
This resource will only affect the local server, correct? I am wondering if it should be mapped under ```/v2/server/logging```

Reply:
Tricky, because the set and remove ops are clustered

Reply:
So `GET /v2/server/logging/loggers` and `GET /v2/server/logging/appenders` but `PUT /v2/cluster/logging/loggers/org.infinispan.ABC` and `DELETE /v2/cluster/logging/loggers/org.infinispan.ABC`

Reply:
WDYM? PUT and DELETE are sent to all nodes, but not GET?

Reply:
PUT and DELETE are handled by clustered tasks.
GET is only handled by the local node. In a symmetrical cluster that should be fine. Making it clustered would mean returning a map of list of loggers (one list per server) which seems overkill